pkg/client/transport.go

```diff
@@ -117,6 +117,28 @@ func TLSConfigFor(config *Config) (*tls.Config, error) {
 	return tlsConfig, nil
 }
 
+// tlsConfigKey returns a unique key for tls.Config objects returned from TLSConfigFor
+func tlsConfigKey(config *Config) (string, error) {
+	hasCA := len(config.CAFile) > 0 || len(config.CAData) > 0
+	hasCert := len(config.CertFile) > 0 || len(config.CertData) > 0
+	if hasCA && config.Insecure {
+		return "", fmt.Errorf("specifying a root certificates file with the insecure flag is not allowed")
+	}
+	if err := LoadTLSFiles(config); err != nil {
+		return "", err
+	}
+	switch {
```

Comment:
only the tls portions of the config are relevant to the key. simplified this to include those portions only. hex-encoded byte arrays are ~1/3 the size of `%#v`, and even less than `spew.Dump`